replica/table.cc

```diff
@@ -1654,9 +1658,13 @@ future<> compaction_group::stop() noexcept {
     if (_async_gate.is_closed()) {
         co_return;
     }
-    co_await _async_gate.close();
+    // Allow `compaction_manager::remove` to stop ongoing compactions
+    // while they may still hold the _async_gate
+    // entered in `compaction_manager::start_compaction`
```

Comment:
ditto

Reply:
no, here it is the compaction_group::_async_gate.
The previous mention was copy-pasted

Reply:
Yes, but it's still a different gate than the one entered in `compaction_manager::start_compaction`

Reply:
You're right.
Dropped this hunk changing to compacton_group::stop in v2